apps/accounts/admin.py

```diff
@@ -13,13 +13,17 @@ class MoodyUserAdmin(MoodyBaseAdmin):
 
 class UserEmotionAdmin(MoodyBaseAdmin):
     form = UpdateUserEmotionAttributesForm
-    list_display = ('user', 'emotion', 'energy', 'valence', 'danceability')
+    list_display = ('user', 'emotion', 'energy', 'valence', 'danceability', 'vote_count_for_emotion')
     readonly_fields = ('user', 'emotion')
     list_filter = ('emotion',)
 
     def has_add_permission(self, request):
         return False
 
+    def vote_count_for_emotion(self, obj):
+        return obj.user.usersongvote_set.filter(emotion=obj.emotion).count()
```

Comment:
This works, but it is not an efficient query. It will perform a DB query for each UserEmotion record in our system, which will not scale for an increased number of users. Napkin math shows that the number of queries for a request to the UserEmotion admin list page is `COUNT(users) x COUNT(emotions)` plus the other queries needed for the page.

Can we add some optimization to the queryset used to `prefetch_related` votes? Another option is moving the vote count display to the info for a specific UserEmotion record instead of in the list page.

Reply:
c7a6a7b adds some clever annotation logic to pull the votes for emotions for each user to the queryset. This seems to be efficient with regards to the number of queries performed.